fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/RelationalExpression.java

```diff
@@ -694,19 +695,22 @@ default Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candi
      * @param aliasMap a map of alias defining the equivalence between aliases and therefore quantifiers
      * @param partialMatchMap a map from quantifier to a {@link PartialMatch} that pulled up along that quantifier
      *        from one of the expressions below that quantifier
+     * @param maxMatchMap the max match map.
```

Comment:
Is this the max match map on this level? How is this constructed? It seems that every caller only ever calls this with `Optional.empty()` though I may be mistaken on this one...

Reply:
You're right, this was done before I reverted it due to a bug that prevents calculation of M3 correctly. I left the signature like this though so I can expand it in a subsequent PR with the proper calculation of the M3.